### ISSUE TITLE
Sample loop points

### ISSUE THREAD

**Issue by jarikomppa** [jarikomppa/soloud#15]
Implement loop points to audio. Currently looping always jumps to the beginning of the stream.

**Comment by ocornut** [comment]
Been looking into this today.

A) In your WavStream looping code for OGG you are doing something like:
```
	if (mOffset >= mParent->mSampleCount)
	{
		if (mFlags & AudioSourceInstance::LOOPING)
		{
			stb_vorbis_seek_start(mOgg);
			mOffset = aSamples - offset;
			mLoopCount++;
		}
```
I don't understand the significance of the line `mOffset = aSamples - offset;` here. 
I think it should say `mOffset = 0 `otherwise the subsequent loop will drift and be triggered earlier, so as I understand it this is a bug in the current library? Same thing apply to WavStream for looped WAV.
I may be completely wrong!

By the way the loop detection logic in the WavStream WAV loop handling is written a little differently from the OGG one when it appears they could be written the same way, so either this is odd either they might be a non-obvious case to document.

B) Below what I'm doing to loop a streamed OGG. I'm not submitting a PR because I don't think this has been properly designed. Right now I am relying on manually submitted loop point. I would assume you would like to extract those from the WAV/OGG point but even if that is possible it would be beneficial to allow the user to programmatically specify one. I can extend this to other types if you want but there's several corner cases to catch, probably you'd be better finisher otherwise I am happy to try to take the idea to commitable level under your design guidance.

Basically adds a `mLoopPointSample` member in WavStream, defaulting to 0, with an setter accessor, and in `WavStreamInstance::getAudio()` fo the OGG path:

```
	if (mParent->mLoopPointSample > 0)
		stb_vorbis_seek(mOgg, mParent->mLoopPointSample);
	else
		stb_vorbis_seek_start(mOgg);
	mOffset = mParent->mLoopPointSample+1;
```

I think `mOffset = mParent->mLoopPointSample` should be correct but on my test OGG I end being stuck at the end of the second music loop with 1 sample missing in the OGG stream (`getOggData` returning 0 and the outer code in getAudio() ends up in an infinite loop). I'm not sure where is the issue, if seeking is off by 1 sample or something?

**Comment by jarikomppa** [comment]
Done in a more generic way now.

**Comment by ocornut** [comment]
Thank you both!

**Comment by jarikomppa** [comment]
I was a bit hesitant to stomp on sammalt's code, but since I found it convenient to add a generic loop point I went with it. This also simplified the code audio sources need to implement in general, so I think it's a win.

What's missing in the more general solution is the loop end point setting that was in sammalt's code.
